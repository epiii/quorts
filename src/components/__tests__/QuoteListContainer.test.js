import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";
import toJSON from "enzyme-to-json";

import ConnectedComponent, { QuoteListContainer } from "../QuoteListContainer";

describe("QuoteListContainer", () => {
  Date.now = jest.fn(() => 1529122840652);

  const mockStore = configureStore();

  const setup = (props = {}, isConnected = true) => {
    const store = mockStore();
    store.dispatch = jest.fn();

    const wrapper = isConnected
      ? shallow(<ConnectedComponent store={store} />)
      : shallow(<QuoteListContainer {...props} />);

    return {
      wrapper,
      store
    };
  };

  it("should renders with quotes", () => {
    const items = [
      {
        id: "1",
        text: "My Quote 1",
        date: Date.now()
      },
      {
        id: "2",
        text: "My Quote 2",
        date: Date.now()
      },
      {
        id: "3",
        text: "My Quote 3",
        date: Date.now()
      }
    ];
    const { wrapper } = setup({ items }, false);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders without quotes", () => {
    const { wrapper } = setup({}, false);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should maps state and dispatch to props", () => {
    const { wrapper } = setup();

    expect(wrapper.props()).toMatchObject({
      items: [],
      onClickRemove: expect.any(Function)
    });
  });

  it("should maps onClickRemove to dispatch remove quote action", () => {
    const { wrapper, store } = setup();

    wrapper.props().onClickRemove(1);

    expect(store.dispatch).toBeCalledWith({
      type: "REMOVE_QUOTE",
      payload: 1
    });
  });
});
