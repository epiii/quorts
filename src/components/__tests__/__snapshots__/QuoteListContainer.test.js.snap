// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteListContainer should renders with quotes 1`] = `
<div>
  <styled.h1>
    Quote List
  </styled.h1>
  <QuoteList
    items={
      Array [
        Object {
          "date": 1529122840652,
          "id": "1",
          "text": "My Quote 1",
        },
        Object {
          "date": 1529122840652,
          "id": "2",
          "text": "My Quote 2",
        },
        Object {
          "date": 1529122840652,
          "id": "3",
          "text": "My Quote 3",
        },
      ]
    }
    onClickCopy={[Function]}
  />
</div>
`;

exports[`QuoteListContainer should renders without quotes 1`] = `
<styled.p>
  There's no quotes
</styled.p>
`;
